.github/workflows/publish.yaml

```diff
@@ -0,0 +1,110 @@
+name: Publish
+on:
+  push:
+    branches:
+      - "main"
+      - v*
+    tags:
+      - v*
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: "go.mod"
+          cache: true
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: "go.mod"
+          cache: true
+
+      - name: Unit Tests
+        run: |
+          make test
+
+      - name: Lint
+        run: |
+          ENV=host make lint
+          make bumplicense
+          go mod tidy
+          pushd e2etests
+          go mod tidy
+          popd
+          make manifests
+          make checkuncommitted
+
+
+  publish-images:
+    runs-on: ubuntu-22.04
+    needs: [unit-tests]
+    permissions:
+      contents: read
+      id-token: write # needed for signing the images with GitHub OIDC Token
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@main
+        with:
+          cosign-release: "v1.13.1"
+
+      - name: Code checkout
+        uses: actions/checkout@v3
+
+      - name: Setup docker buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log into Quay
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            quay.io/metallb/frr-k8s
+          # generate Docker tags based on the following events/attributes
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{raw}}
+          labels: |
+            org.opencontainers.image.title=frr-k8s
+            org.opencontainers.image.description=frr-k8s, a cloud native wrapper of some frr features
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        id: build-and-push
+        with:
+          context: .
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          file: ${{matrix.image}}/Dockerfile
```

Comment:
I don't think we define the matrix here like we do in metallb

Reply:
Yep, leftover. Removed